src/lat/lattice-functions.h

```diff
@@ -306,6 +306,21 @@ void ComposeCompactLatticeDeterministic(
     fst::DeterministicOnDemandFst<fst::StdArc>* det_fst,
     CompactLattice* composed_clat);
 
+/// This function returns true if the CompactLattice clat has unique
+/// left phone context of length n, and false if it does not.
+bool HasUniquePhoneContext(const CompactLattice &clat,
+       const TransitionModel &tmodel, int32 n);
```

Comment:
this function HasUniquePhoneContext and also CompactLatticeExpandByPhone should not be needed- composition will take care of that.  Make sure you understand how FST composition works.  It keeps a queue of (left-state, right-state) pairs and maintains a map (left-state, right-state) -> composed-state.